### ISSUE TITLE
Bug: NPE when applying layout

### ISSUE THREAD

**Issue by tlf30** [andrewauclair/ModernDocking#61]
Hello,
I'm not sure what changed, but recently when I start my app, I now get a NullPointerException applying the layout.
Here is the layout and exception:
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "ModernDocking.layouts.ApplicationLayout.getMainFrameLayout()" because "layout" is null
	at ModernDocking.DockingState.restoreApplicationLayout(DockingState.java:94)
	at ModernDocking.persist.AppState.restore(AppState.java:121)
	at io.tlf.cip.tool.ui.MainWindow.lambda$restoreLayout$0(MainWindow.java:68)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```
```xml
<?xml version="1.0" ?>
<app-layout>

<layout main-frame="true" location="-7,-7" size="1721,1033" state="6">
<westToolbar>
</westToolbar>
<eastToolbar>
</eastToolbar>
<southToolbar>
</southToolbar>
<split orientation="1" divider-proportion="0.2497255802154541">
<left>
<simple persistentID="Message Log">
<properties dividerLocation="165"></properties>

</simple>
</left>
<right>
<split orientation="1" divider-proportion="0.0">
<left>
<simple persistentID="Message Log">
<properties dividerLocation="165"></properties>

</simple>
</left>
<right>
<split orientation="1" divider-proportion="0.11348562687635422">
<left>
<simple persistentID="Discovery">
<properties orientation="0" dividerLocation="724"></properties>

</simple>
</left>
<right>
<tabbed>
<selectedTab persistentID="Message">
</selectedTab>
<tab persistentID="Identity">
</tab>
<tab persistentID="Email">
</tab>
<tab persistentID="Controller">
</tab>
<tab persistentID="Message">
</tab>
</tabbed>
</right>
</split>
</right>
</split>
</right>
</split>
</layout>
</app-layout>
```

``Java
public MainWindow(Main main) {
        this.main = main;

        initDocking();

        //Init JFormDesigner
        initComponents();

        initTools();

        //Restore docking layout
        restoreLayout();
    }

    private void initDocking() {
        // create the single Docking instance for the life of the program
        Docking.initialize(this);

        // set the auto persist file
        // auto persist is off by default and we will leave it that way until we're done configuring the default layout
        // that way we're not persisting the default layout and overwriting the file before we load the file
        AppState.setPersistFile(new File("layout.xml"));
    }

    private void restoreLayout() {
        // restore the layout from the auto persist file after the UI has loaded
        SwingUtilities.invokeLater(() -> {
            AppState.restore();
            // now that we've restored the layout we can turn on auto persist
            AppState.setAutoPersist(true);
        });
    }
```

By removing the layout.xml the issue goes away, so I assume it is something in the layout file that is corrupted.
If I generate a new layout file, it will continue to work.

**Comment by andrewauclair** [comment]
I added a way to set the default app layout in `AppState`. Looks like I'm missing some null checks to see if that layout actually exists.

I'm also missing some logging which is hiding your actual exception. If you add `e.printStackTrace();` to AppState in the catch at line 120, you should see the real exception.

**Comment by tlf30** [comment]
This seems to be fixed in 0.6.0. I have been doing extensive testing after updating and not getting this exception anymore.
